scripts/linkbot/monitor-raspbian-os.js

```diff
@@ -70,46 +21,40 @@ function updateGitlabCILink(newLine, newURL, newName) {
     console.error(err);
     process.exit(1);
   }
-}
-
-JSDOM.fromURL(url, {}).then((dom) => {
-  var document = dom.window.document;
-  var table = document.getElementsByTagName("table");
-  var rows = table[0].rows;
-  var matches = [];
-  for (var i = 0; i < rows.length; i++) {
-    var rowText = rows[i].textContent;
-    var regMatch = rowText.match(reg);
-    if (regMatch) {
-      matches.push(regMatch);
-    }
-  }
-  // Sort the accumulated matches
-  matches.sort(function (a, b) {
-    let al = Date.parse(a.groups.date);
-    let bl = Date.parse(b.groups.date);
-    if (al == bl) {
-      let ad = Date.parse(a.groups.updated);
-      let bd = Date.parse(b.groups.updated);
-      return bd - ad;
+};
+
+const checkForUpdates = async (url, currentImageName) =>
+  getLinksByMatch(url, reg).then(async (links) => {
+    const { link, match } = links.sort((a, b) => Date.parse(b.match.groups.date) - Date.parse(a.match.groups.date))[0];
+    const matchOn = match.input.split('/')[0];
+    if (matchOn !== currentImageName) {
+      // We also need to extract the new image name from the index folder, as
+      // dated folders contain images with different dates in them o_O
+      console.error("We've got a new release! \\o/");
+      // Update the GitlabCI link unconditionally
+      console.log('Updating the GitlabCI Link');
+      const { newLink, imageName } = await getLinksByMatch(link, 'raspios-.*-lite.*.img.(zip|xz)$').then((links) => {
+        const { link, match } = links[0];
+        const imageName = match.input.substring(0, match.input.lastIndexOf('.img'));
+        return { newLink: link, imageName };
+      });
+      console.log(`New release: ${newLink}`);
+      updateGitlabCILink({ name: 'RASPBIAN_NAME', url: 'RASPBIAN_URL' }, imageName, newLink);
     }
-    return bl - al;
   });
-  var matchOn = matches[0].input.split("/")[0];
-  if (matchOn !== imageName) {
-    // We also need to extract the new image name from the index folder, as
-    // dated folders contain images with different dates in them o_O
-    console.error("We've got a new release! \\o/");
 
-    // Update the GitlabCI link unconditionally
-    console.log("Updating the GitlabCI Link");
-    var newVar = getNewRaspbian(
-      "RASPBIAN_URL",
-      `${url}${matches[0].input.split(" ")[0].split("/").slice(0, -1)}`,
-      (separator = ":")
-    ).then(({ replacementLine, imageName }) => {
-      console.log(`New release: ${replacementLine}`);
-      updateGitlabCILink(replacementLine, "RASPBIAN_URL", imageName);
-    });
-  }
-});
+// Read the input file, and parse the variable input
```

Comment:
Q: If the above is `updateCILink`. Should this be `GetCILink`?

Reply:
getting this in line too!